pyiceberg/catalog/__init__.py

```diff
@@ -754,9 +760,7 @@ def _load_file_io(self, properties: Properties = EMPTY_DICT, location: Optional[
         return load_file_io({**self.properties, **properties}, location)
 
     @staticmethod
-    def _convert_schema_if_needed(schema: Union[Schema, "pa.Schema"]) -> Schema:
-        if isinstance(schema, Schema):
-            return schema
+    def _convert_to_iceberg_schema(schema: "pa.Schema") -> Schema:
```

Comment:
It could be that folks are using this method, I would refrain from changing it when not needed.

Reply:
@Fokko I changed it because I needed to determine if the schema needed to be converted outside of this function. Based on whether or not the schema had been converted (re-ordered), I'm making the determination that fresh IDs need to be assigned.

For instance: https://github.com/apache/iceberg-python/pull/1304/files#diff-3bda7391ebd8aa3dcfd6703d8d2764830b9d9c35fa854188a37d69611274bd3dR576-R584